ibis/backends/base/__init__.py

```diff
@@ -322,8 +322,52 @@ def to_pyarrow_batches(
         """
         raise NotImplementedError
 
+    def read_parquet(
+        self, path: str | Path, table_name: str | None = None, **kwargs: Any
+    ) -> ir.Table:
+        """Register a parquet file as a table in the current backend.
 
-class BaseBackend(abc.ABC, ResultHandler):
+        Parameters
+        ----------
+        path
+            The data source.
+        table_name
+            An optional name to use for the created table. This defaults to
+            a sequentially generated name.
+        **kwargs
+            Additional keyword arguments passed to the backend loading function.
+
+        Returns
+        -------
+        ir.Table
+            The just-registered table
+        """
+        raise NotImplementedError
```

Comment:
It can be informative to report the name of the specific backend that raised this expression in case this is not the one the user would have expected to make it easier for them to updated their install / runtime config / code accordingly.


```suggestion
        raise NotImplementedError(
            f"{self.__class__} does not support direct registration of parquet data."
        )
```